### DISCUSSION TITLE
'ProgressBar' object has no attribute 'main_progress_bar'

### DISCUSSION THREAD

**Discussion by marwanmusa** [ourownstory/neural_prophet#1218] [Q&A - get help using NeuralProphet] [answered]
I got this error:
Trainer.__init__() got an unexpected keyword argument 'auto_lr_find'

setting learning_rate to a value fix that error above. Actually, previously I don't have to set the learning rate (before a try to update the package this morning).

Then, after setting the lr, error below exists:
what did I do wrong? or there are some changes that I missed?

environment Jupyter.

I have checked in utils.py
there is an import 'ProgressBar' from neuralprophet.logger

in logger.py class ProgressBar that inherits from TQDMProgressBar
I can't find main_progress_bar attribute.
🙏🏻
can anybody help...

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
Cell In[5], line 17
     12 # remove some last periods, create test dataset
     13 # train = data_pre.drop(data_pre.index[-(10):])
     14 
     15 # define the model
     16 model = NeuralProphet()
---> 17 metrics = model.fit(data_pre, freq='D', learning_rate=0.001, progress='print')
     19 # # df_future = model.make_future_dataframe(data_pre, periods=365)
     20 # forecast = model.predict(data_pre)
     21 
   (...)
     53 # plt.legend()
     54 # plt.show()

File [d:\Projects_assignment\gitlab\financial-forecasting\.venv\lib\site-packages\neuralprophet\forecaster.py:866](file:///D:/Projects_assignment/gitlab/financial-forecasting/.venv/lib/site-packages/neuralprophet/forecaster.py:866), in NeuralProphet.fit(self, df, freq, validation_df, epochs, batch_size, learning_rate, early_stopping, minimal, metrics, progress, checkpointing, continue_training, num_workers)
    864 # Training
    865 if validation_df is None:
--> 866     metrics_df = self._train(
    867         df,
    868         progress_bar_enabled=bool(progress),
    869         metrics_enabled=bool(self.metrics),
    870         checkpointing_enabled=checkpointing,
    871         continue_training=continue_training,
...
---> 57     self.main_progress_bar.reset(self.epochs)
     58     self.main_progress_bar.set_description(f"Epoch {trainer.current_epoch + 1}")
     59     self._update_n(self.main_progress_bar, trainer.current_epoch + 1)

AttributeError: 'ProgressBar' object has no attribute 'main_progress_bar'

**Comment by bernasor** [comment]
I got this today also and I'm stuck

**Comment by noxan** [comment, marked as answer]
@marwanmusa @bernasor we released a new version of NeuralProphet (`0.5.3`) which should resolve those issues.
